packages/test/stochastic-test-utils/src/performActions.ts

```diff
@@ -99,30 +100,51 @@ export async function performFuzzActionsAsync<
 	const applyOperation: (operation: TOperation) => Promise<TState> = async (op) =>
 		(await reducer(state, op)) ?? state;
 
+	let occurredFirstWrite = false;
```

Comment:
The fact that we re-throw the error when there is one means we'll only ever save one file for a given run. That means `occurredFirstWrite` will never be used after being set to `true`.
I recommend just removing the logic associated with it. `mkdir` doesn't overwrite anyway.